src/python/pants/backend/java/compile/javac.py

```diff
@@ -154,51 +171,28 @@ async def compile_java_source(
             for t in component_members_with_sources
         ),
     )
-
     component_members_and_java_source_files = [
         (target, sources)
         for target, sources in component_members_and_source_files
         if sources.snapshot.digest != EMPTY_DIGEST
     ]
-
     if not component_members_and_java_source_files:
+        # If the component has no sources, it is acting as an alias for its dependencies: return
+        # their merged classpaths.
```

Comment:
What does this represent? An explicit dep on a `java_sources` target generator?

Reply:
Any target type with no matching sources: so yea, `java_sources`, lockfile targets, etc.